### ISSUE TITLE
Is there a way to use the tmp file created after filling a pdf to add to another PDF?

### ISSUE THREAD

**Issue by lumbergh** [mikehaertl/php-pdftk#137]
$files = [
	    [  
	        'name' => 'formA.pdf',
	        'data' => array('fieldA'=>'some value', 'fieldB'=>'another value', 'fieldC'=>'yet another value')
	    ],
	    [  
	        'name' => 'formB.pdf',
	        'data' => array('fieldA'=>'this value', 'fieldB'=>'that value', 'fieldC'=>'no value')
	    ]
	];      

     // Main PDF
	$pdf = new Pdf();  

	foreach($files as $file) {
	    $tmp = new Pdf($file['name']);
	    $tmp->fillForm($file['data'])->execute();
	    $pdf->addFile((string)$tmp->getTmpFile());
	}

	if (!$pdf->saveAs(sys_get_temp_dir().'/merged.pdf')) {
	    die($pdf->getError());
	}

I might be wrong but It seems my main problem is coming from the tmp file disappearing before I can add it to the main pdf.  Is there a better way to do this?  

I didn't want to have to execute  a saveAs on the individual pdfs to improve performance but I'm not sure that it can actually work like this now.

**Comment by mikehaertl** [comment]
@lumbergh You're probably right. We use destructors in our `php-tmpfile` class to clean up:

https://github.com/mikehaertl/php-tmpfile/blob/master/src/File.php#L56

Destructors get  called, as soon as the instance is no longer referenced in the code. I've already tried to fix this but didn't consider that you may call `addFile()` multiple times. 

Could you do me a favour and try something?

Can you change the following lines in our `Pdf.php` class:

```
protected $_pdf;
// change to:
protected $_pdfs = array();

// in function addFile()
$this->_pdf = $name;
// change to:
$this->_pdfs[] = $name;
```

**Comment by lumbergh** [comment]
Unfortunately doesn't seemed to have prevented them from being removed.  At least in my local test on MacOS.  If I get more time to look into it I will investigate further.  
Thanks for your quick response. Keep up the great work on the library!

**Comment by mikehaertl** [comment]
Too bad. Maybe you can add some debugging/logging here and there. I'd try to figure out, in which order the `__destructor()` methods in the php-tmpfile get called or rather if they get called too early.

**Comment by mikehaertl** [comment]
Closing this as there was no further feedback.

**Comment by jamespavett** [comment]
@mikehaertl I do have this exact same issue. When adding files in a foreach loop, the tmp PDFs seems seem to be deleted by the time the new PDF is ready to be saved.

For now I am going to try and save the files between each iteration, but would be great if it was achievable through accessing the tmp files. I could potentially investigate further to see when the destructors are called?

**Comment by mikehaertl** [comment]
@jamespavett Have you tried my above suggestion? If we keep a reference to each file in an `$_pdfs` array this should prevent the destructor from getting called.

**Comment by mikehaertl** [comment]
@jamespavett One more thing: The suggested change only works if you pass the `Pdf` object to `addFile()`, not the filename of a temp file. Actually some (minimal) example how you use it would help.

**Comment by jamespavett** [comment]
@mikehaertl so the suggestion from above regarding `$_pdfs` seemed to work perfectly. Now it can read the tmp files after the foreach loop as completed.

I am using it in the below way. Filling a form on each iteration of the loop and adding it back onto the original PDF.

```
   $pdf = new Pdf();
    foreach (range(1, 8) as $number) {

        $sheet = new Pdf(storage_path().'/app/pdfs/Sheets.pdf');
        $sheet->fillForm([
            'counterRow1' => 'test' . $number,
        ])->flatten();

        $pdf->addFile($sheet);
    }

    $result = $pdf->needAppearances()->saveAs('filled-in-sheet.pdf');
```

As a temporary workaround I have just been doing the following, saving a new copy on each iteration:

 ```
   $pdf = new Pdf();
    foreach (range(1, 90) as $number) {

        $sheet = new Pdf(storage_path().'/app/pdfs/DriverSheets.pdf');
        $sheet->fillForm([
            'Trade counterRow1' => 'test' . $number,
        ])->flatten()->needAppearances()->saveAs('filled-in-sheet' . '-' . $number . '.pdf');

        $pdf->addFile(public_path(). '/filled-in-sheet' . '-' . $number . '.pdf');
    }

    $result = $pdf->needAppearances()->saveAs('filled-in-sheet.pdf');
```

Is this is something that could get fixed via a release? Or would the best thing to do just be extend the Pdf class and override the property and addFile methods?

**Comment by mikehaertl** [comment]
@jamespavett Thanks for your help. I've just released 0.10.1 containing this fix.

**Comment by jamespavett** [comment]
@mikehaertl Thank you so much for the quick resolution!
